var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -176,37 +188,46 @@ class Qt(Package):
     conflicts("%apple-clang@13:", when="@:5.13")
 
     # Build-only dependencies
-    depends_on("pkgconfig", type="build")
+    for plat in ["linux", "darwin", "freebsd"]:
+        with when(f"platform={plat}"):
+            depends_on("pkgconfig", type="build")
+            depends_on("libsm", when="@3")
+            depends_on("glib", when="@4:")
+            depends_on("libmng")
+            depends_on("assimp@5.0.0:5", when="@5.5:+opengl")
+            depends_on("sqlite+column_metadata", when="+sql", type=("build", "run"))
+    # Windows sqlite has no column_metadata variant
+    with when("platform=windows +sql"):
+        depends_on("sqlite", type=("build", "run"))
+
+    depends_on("icu4c")
+    depends_on("harfbuzz", when="@5:")
+
     depends_on("python", when="@5.7.0:", type="build")
 
     # Dependencies, then variant- and version-specific dependencies
-    depends_on("icu4c")
     depends_on("jpeg")
-    depends_on("libmng")
     depends_on("libtiff")
     depends_on("libxml2")
     depends_on("zlib-api")
     depends_on("freetype", when="+gui")
     depends_on("gtkplus", when="+gtk")
-    depends_on("sqlite+column_metadata", when="+sql", type=("build", "run"))
 
     depends_on("libpng@1.2.57", when="@3")
-    depends_on("libsm", when="@3")
     depends_on("pcre+multibyte", when="@5.0:5.8")
-    depends_on("inputproto", when="@:5.8")
+    for plat in ["linux", "darwin", "freebsd"]:
+        with when(f"platform={plat}"):
+            depends_on("inputproto", when="@:5.8")
```

Comment:
IMO it would make sense to consolidate this into the initial platform loop you added above.